File: app/src/main/java/com/jakebarnby/filemanager/ui/dialogs/SettingsDialog.java
package com.jakebarnby.filemanager.ui.dialogs;

import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;
import android.view.View;
import android.widget.CheckBox;

import com.jakebarnby.filemanager.R;
import com.jakebarnby.filemanager.sources.SourceActivity;
import com.jakebarnby.filemanager.util.Constants;
import com.jakebarnby.filemanager.util.PreferenceUtils;

/**
 * Created by Jake on 10/3/2017.
 */

public class SettingsDialog extends DialogFragment {

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle(getString(R.string.action_settings));

        View rootView = getActivity().getLayoutInflater().inflate(R.layout.dialog_settings, null);
        initViews(rootView);

        builder.setView(rootView);
        builder.setNegativeButton(getString(R.string.close), (dialog, which) -> {
            ((SourceActivity)getActivity()).initAllRecyclers();
            dialog.dismiss();
        });

        return builder.create();
    }

    private void initViews(View rootView) {
        CheckBox foldersFirst = rootView.findViewById(R.id.ckb_folders_first);
        CheckBox hiddenFolders = rootView.findViewById(R.id.ckb_hidden_files);

        boolean bFoldersFirst = PreferenceUtils.getBoolean(getContext(), Constants.Prefs.FOLDER_FIRST_KEY, true);
        boolean bShowHiddenFiles = PreferenceUtils.getBoolean(getContext(), Constants.Prefs.HIDDEN_FOLDER_KEY, false);

        foldersFirst.setChecked(bFoldersFirst);
        hiddenFolders.setChecked(bShowHiddenFiles);

        foldersFirst.setOnCheckedChangeListener((compoundButton, checked) -> {
            PreferenceUtils.savePref(getContext(), Constants.Prefs.FOLDER_FIRST_KEY, checked);
        });

        hiddenFolders.setOnCheckedChangeListener((compoundButton, checked) -> {
            PreferenceUtils.savePref(getContext(), Constants.Prefs.HIDDEN_FOLDER_KEY, checked);
        });

//        Button ossButton = rootView.findViewById(R.id.btn_oss_licenses);
//        ossButton.setOnClickListener((view) -> {
//            Intent intent = new Intent(getActivity(), OssLicensesMenuActivity.class);
//            String title = getString(R.string.oss_license_title);
//            intent.putExtra(Constants.DIALOG_TITLE_KEY, title);
//            startActivity(intent);
//
//        });
    }
}

File: app/src/main/java/com/jakebarnby/filemanager/ui/dialogs/SettingsDialog.kt
package com.jakebarnby.filemanager.ui.dialogs

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.view.View
import android.widget.CheckBox
import androidx.appcompat.app.AlertDialog
import androidx.core.os.bundleOf
import androidx.fragment.app.DialogFragment
import com.jakebarnby.filemanager.R
import com.jakebarnby.filemanager.managers.PreferenceManager
import com.jakebarnby.filemanager.util.Constants.DIALOG_ON_POSITIVE_KEY
import com.jakebarnby.filemanager.util.Constants.DIALOG_TITLE_KEY
import com.jakebarnby.filemanager.util.Constants.Prefs

/**
 * Created by Jake on 10/3/2017.
 */
class SettingsDialog : DialogFragment() {

    companion object {
        fun newInstance(
            onPositive: () -> Unit
        ) =
            SettingsDialog().apply {
                arguments = bundleOf(
                    DIALOG_TITLE_KEY to getString(R.string.action_settings),
                    DIALOG_ON_POSITIVE_KEY to onPositive
                )
            }
    }

    private val prefs = PreferenceManager(context!!.getSharedPreferences(
        Prefs.PREFS,
        Context.MODE_PRIVATE
    ))

    private var hiddenFilesChecked = false
    private var foldersFirstChecked = false

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val onPositive = arguments
            ?.getSerializable(DIALOG_ON_POSITIVE_KEY) as? () -> Unit

        val builder = AlertDialog.Builder(activity!!).setTitle(getString(R.string.action_settings))

        val view = layoutInflater.inflate(R.layout.dialog_settings, null)
        initViews(view)

        return builder
            .setView(view)
            .setPositiveButton(R.string.ok) { _, _ ->
                prefs.savePref(Prefs.FOLDER_FIRST_KEY, foldersFirstChecked)
                prefs.savePref(Prefs.HIDDEN_FILES_KEY, hiddenFilesChecked)
                onPositive?.invoke()
            }
            .setNegativeButton(R.string.close) { _, _ ->
                dismiss()
            }.create()
    }

    private fun initViews(rootView: View) {
        val foldersFirstCbx = rootView.findViewById<CheckBox>(R.id.ckb_folders_first)
        val hiddenFoldersCbx = rootView.findViewById<CheckBox>(R.id.ckb_hidden_files)
        val foldersFirst = prefs.getBoolean(Prefs.FOLDER_FIRST_KEY, true)
        val showHiddenFiles = prefs.getBoolean(Prefs.HIDDEN_FILES_KEY, false)

        foldersFirstCbx.isChecked = foldersFirst
        hiddenFoldersCbx.isChecked = showHiddenFiles

        foldersFirstCbx.setOnCheckedChangeListener { _, checked ->
            foldersFirstChecked = checked
        }
        hiddenFoldersCbx.setOnCheckedChangeListener { _, checked ->
            hiddenFilesChecked = checked
        }
    }
}